mongodb/vibe/db/mongo/sasl.d

```diff
@@ -0,0 +1,153 @@
+/**
+	SASL authentication functions
+
+	Copyright: © 2012-2016 Nicolas Gurrola
+	License: Subject to the terms of the MIT license, as written in the included LICENSE.txt file.
+	Authors: Nicolas Gurrola
+*/
+module vibe.db.mongo.sasl;
+
+import std.algorithm;
+import std.base64;
+import std.conv;
+import std.digest.hmac;
+import std.digest.sha;
+import std.exception;
+import std.format;
+import std.string;
+import std.utf;
+import vibe.crypto.cryptorand;
+
+private SHA1HashMixerRNG g_rng;
+
+struct ScramState
```

Comment:
I'd mark this as `package`, so that we don't have to worry about the API so much for now.